### ISSUE TITLE
entityManager ORDER BY CONCAT strips out strings

### ISSUE THREAD

**Issue by jodysalt** [icatproject/icat.server#172]
If try to run the following (on a 4.6.1 icat):

select distinct investigation from Investigation investigation , investigation.facility as facility where facility.id = 1 and investigation.name = 'APPLIEDAHEAD' and investigation.startDate between {ts 0000-01-01 00:00:00} and {ts 2016-05-13 23:59:59} ORDER BY CONCAT(investigation.name, '-', investigation.visitId) asc limit 0, 50

I get the folowing response:

{"code":"BAD_PARAMETER","message":"Timestamp literals have been replaced... An exception occurred while creating a query in EntityManager: \nException Description: Syntax error parsing [SELECT DISTINCT investigation FROM Investigation investigation , investigation.facility AS facility WHERE facility.id = 1 AND investigation.name = APPLIEDAHEAD AND investigation.startDate BETWEEN  CURRENT_TIMESTAMP  AND  CURRENT_TIMESTAMP  ORDER BY CONCAT ( investigation.name , - , investigation.visitId ) ASC]. \n[273, 273] An arithmetic factor must be followed by an expression."}

Basically you can see the '-' has had it's quotes stripped out some how to it gets treated as as minus operator.

**Comment by fisherab** [comment]
I believe that this is not valid JPQL. http://docs.oracle.com/html/E13946_02/ejb3_langref.html#ejb3_langref_orderby says that an orderby is based on a state_field_path_expression which does not appear to allow string functions such as CONCAT.

**Comment by jodysalt** [comment]
According to the spec you're right - though I am almost positive it has worked in the past (though it shouldn't have).

I can get around it though by performing the ordering on a vaiable defined in the select part of theb statement.
